sdk/iot/Azure.Iot.Hub.Service/tests/ConfigurationsClientTests.cs

```diff
@@ -0,0 +1,150 @@
+                Response<TwinConfiguration> createResponse =
+                    await client.Configurations.CreateOrUpdateConfigurationAsync(twinConfiguration);
+                createdConfig = createResponse.Value;
+
+                // Get twin configuration
+                Response<TwinConfiguration> getResponse = await client.Configurations.GetConfigurationAsync(testConfigurationId).ConfigureAwait(false);
+
+                getResponse.Value.Etag.Should().BeEquivalentTo(createdConfig.Etag, "ETag value should not have changed.");
+                createdConfig = createResponse.Value;
+
+                // Update a configuration
+                createdConfig.Priority = testPriority;
+                Response<TwinConfiguration> updatedConfig = await client.Configurations.CreateOrUpdateConfigurationAsync(createdConfig, IfMatchPrecondition.UnconditionalIfMatch).ConfigureAwait(false);
+                Assert.AreEqual(updatedConfig.Value.Priority, testPriority, "Priority should have been updated.");
+            }
+            finally
+            {
+                // Delete twin configuration
+                await Cleanup(client, twinConfiguration);
+            }
+        }
+
+        /// <summary>
+        /// Test get configurations
+        /// </summary>
+        [Test]
+        public async Task ConfigurationsClient_GetConfigurations()
+        {
+            const int configurationsCount = 5;
+            TwinConfiguration[] twinConfigurations = new TwinConfiguration[configurationsCount];
+            TwinConfiguration[] createdConfigurations = new TwinConfiguration[configurationsCount];
+            IReadOnlyList<TwinConfiguration> listConfigurations;
+            IotHubServiceClient client = GetClient();
+            try
+            {
+                for (int i = 0; i < configurationsCount; i++)
+                {
+                    twinConfigurations[i] = CreateTestConfig($"testconfigurations{i}{GetRandom()}");
+
+                    // Create Configurations
+                    Response<TwinConfiguration> createResponse =
+                        await client.Configurations.CreateOrUpdateConfigurationAsync(twinConfigurations[i]).ConfigureAwait(false);
+                    createdConfigurations[i] = createResponse.Value;
+                }
+
+                // List the configurations for the client
+                listConfigurations = (await client.Configurations.GetConfigurationsAsync().ConfigureAwait(false)).Value;
+
+                IEnumerable<string> twinConfigurationsIds = listConfigurations.ToList().Select(configuration => configuration.Id);
+
+                // Compare the response ids with created configurations
+                for (int i = 0; i < configurationsCount; i++)
+                {
+                    Assert.IsTrue(twinConfigurationsIds.Contains(twinConfigurations[i].Id));
+                }
+            }
+            finally
+            {
+                for (int i = 0; i < configurationsCount; i++)
```

Comment:
Is there any bulk API to delete all instead of a loop? Also, move the loop to the cleanup method.

Reply:
we do not have a bulk delete api for configs